editor/draw.js

```diff
@@ -407,22 +425,36 @@ svgedit.draw.Drawing.prototype.identifyLayers = function() {
 		// TODO(codedread): What about internationalization of "Layer"?
 		while (layernames.indexOf(("Layer " + i)) >= 0) { i++; }
 		var newname = "Layer " + i;
-		a_layer = svgdoc.createElementNS(NS.SVG, "g");
-		var layer_title = svgdoc.createElementNS(NS.SVG, "title");
-		layer_title.textContent = newname;
-		a_layer.appendChild(layer_title);
+		a_layer = this._doCreateLayer(newname);
 		var j;
 		for (j = 0; j < orphans.length; ++j) {
 			a_layer.appendChild(orphans[j]);
 		}
-		this.svgElem_.appendChild(a_layer);
-		this.all_layers.push( [newname, a_layer] );
+		this.all_layers.push([newname, a_layer] );
 	}
 	svgedit.utilities.walkTree(a_layer, function(e){e.setAttribute("style", "pointer-events:inherit");});
 	this.current_layer = a_layer;
 	this.current_layer.setAttribute("style", "pointer-events:all");
 };
 
+	/**
+	 * Private function that actually creates a new top-level layer in the drawing
+	 * with the given name and sets the current layer to it.
+	 * @param {string} name - The given name
+	 * @returns {SVGGElement} The SVGGElement of the new layer, which is
+	 * also the current layer of this drawing.
+	 */
+	svgedit.draw.Drawing.prototype._doCreateLayer = function(name) {
+		var svgdoc = this.svgElem_.ownerDocument;
+		var new_layer = svgdoc.createElementNS(NS.SVG, "g");
+		addLayerClass(new_layer);
+		var layer_title = svgdoc.createElementNS(NS.SVG, "title");
+		layer_title.textContent = name;
+		new_layer.appendChild(layer_title);
+		this.svgElem_.appendChild(new_layer);
+		return new_layer;
+	};
```

Comment:
This is starting to feel like the concept of a Layer deserves to be its own class.  I did a little sniffing around in this module and sure enough I made a Layer class and then never used it:  svgedit.draw.Layer exists at the top of this file :-/

Can you add the code to create a layer into the Layer constructor and then use it in some way?  Or is this too much refactoring to bite off right now?

I even see a TODO I added for the all_layers array:  "Turn this into an Array.<Layer>"

sigh... sorry about the mess!